prow/create-build-cluster.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+# This script is used to create a new build cluster for use with oss-prow. The cluster will have a 
+# single pd-ssd nodepool that will have autoupgrade and autorepair enabled.
+#
+# Usage: populate the parameters by setting them below or specifying environment variables then run
+# the script and follow the prompts. You'll be prompted to share some credentials and commands
+# with the current oncall.
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Specific to Prow instance
+PROW_INSTANCE_NAME="${PROW_INSTANCE_NAME:-}"
+GCS_BUCKET="${GCS_BUCKET:-${PROW_INSTANCE_NAME}}"
+
+# Specific to the build cluster
+TEAM="${TEAM:-}"
+PROJECT="${PROJECT:-${PROW_INSTANCE_NAME}-build-${TEAM}}"
+ZONE="${ZONE:-us-west1-b}"
+CLUSTER="${CLUSTER:-${PROJECT}}"
+
+# Only needed for creating cluster
+MACHINE="${MACHINE:-n1-standard-8}"
+NODECOUNT="${NODECOUNT:-5}"
+DISKSIZE="${DISKSIZE:-100GB}"
+
+# Only needed for creating project
+FOLDER_ID="${FOLDER_ID:-0123}"
+BILLING_ACCOUNT_ID="${BILLING_ACCOUNT_ID:-0123}"  # Find the billing account ID in the cloud console.
+
+# Overriding output
+OUT_FILE="${OUT_FILE:-build-cluster-kubeconfig.yaml}"
+
+function main() {
+  parseArgs "$@"
+  prompt "Create project" createProject
+  prompt "Create cluster" createCluster
+  prompt "Create a SA and secret for uploading results to GCS" createUploadSASecret
+  prompt "Generate kubeconfig credentials for Prow" gencreds
+  echo "All done!"
+}
+# Prep and check args.
+function parseArgs() {
+  for var in TEAM PROJECT ZONE CLUSTER MACHINE NODECOUNT DISKSIZE FOLDER_ID BILLING_ACCOUNT_ID; do
+    if [[ -z "${!var}" ]]; then
+      echo "Must specify ${var} environment variable (or specify a default in the script)."
+      exit 2
+    fi
+    echo "${var}=${!var}"
+  done
+}
+function prompt() {
+  local msg="$1" cmd="$2"
+  echo
+  read -r -n1 -p "$msg ? [y/n] "
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    "$cmd"
+  else
+    echo "Skipping and continuing to next step..."
+  fi
+}
+function pause() {
+  read -n 1 -s -r
+}
+
+authed=""
+function getClusterCreds() {
+  if [[ -z "${authed}" ]]; then
+    gcloud container clusters get-credentials --project="${PROJECT}" --zone="${ZONE}" "${CLUSTER}"
+    authed="true"
+  fi
+}
+function createProject() {
+  # Create project, configure billing, enable GKE, add IAM rule for oncall team.
+  echo "Creating project '${PROJECT}' (this may take a few minutes)..."
+  gcloud projects create "${PROJECT}" --name="${PROJECT}" --folder="${FOLDER_ID}"
+  gcloud beta billing projects link "${PROJECT}" --billing-account="${BILLING_ACCOUNT_ID}"
+  gcloud services enable "container.googleapis.com" --project="${PROJECT}"
+  gcloud projects add-iam-policy-binding "${PROJECT}" --member="group:mdb.cloud-kubernetes-engprod-oncall@google.com" --role="roles/owner"
```

Comment:
should `member` here  be configurable? Or is the script intended for google-managed clusters only?

Reply:
It should definitely be configurable. Fixed.
I left `group:mdb.cloud-kubernetes-engprod-oncall@google.com` as the default to provide an example value.